source/adios2/transform/compress/CompressBZip2.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressBZip2.h : wrapper to BZip2 compression library
+ *
+ *  Created on: Jul 24, 2017
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#ifndef ADIOS2_TRANSFORM_COMPRESSION_COMPRESSBZIP2_H_
+#define ADIOS2_TRANSFORM_COMPRESSION_COMPRESSBZIP2_H_
+
+#include <bzlib.h>
+
+#include "adios2/core/Transform.h"
+
+namespace adios2
+{
+namespace transform
+{
+
+class CompressBZip2 : public Transform
+{
+
+public:
+    /**
+     * Unique constructor
+     * @param debugMode
+     */
+    CompressBZip2(const bool debugMode);
+
+    ~CompressBZip2() = default;
+
+    size_t BufferMaxSize(const size_t sizeIn) const final;
```

Comment:
See my comment about it. We should mark it final as it's truly final today. We can always change that (it's literally replacing text) when an actual implementation exists for extending this class.